### PR TITLE
Clean up NCBI disease dataset

### PR DESCRIPTION
**Pull request by JetRunner**
This PR cleans up NCBI disease dataset.

- One template that is not doable is removed.
- More templates for the original tasks are added.
- Metrics added.

### CODE REVIEW

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 1]
```diff
@@ -83,8 +84,54 @@ templates:
...
+      {% endif %}
+
+      {% endfor %}
+
+      {{diseases.list | join | trim(",") if (diseases.list | length) > 0 else "None"}}'
+    metadata: !TemplateMetadata
+      choices_in_prompt: false
+      metrics:
+      - Accuracy
+      - Other
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:92]:
```suggestion
      - Other
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 2]
```diff
@@ -83,8 +84,54 @@ templates:
       original_task: false
     name: simple_question_asking_token_indexes_list
     reference: A simple question asking which token indexes are speaking about a disease
-  5a693a8f-07a2-4d88-ab3a-337b1029d9a2: !Template
+  4e96f535-07d4-4c71-8816-3c1cb1900090: !Template
     answer_choices: null
+    id: 4e96f535-07d4-4c71-8816-3c1cb1900090
+    jinja: 'Identify the names of diseases mentioned in the following text:
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:53]:
```suggestion
    jinja: 'Identify the names of diseases mentioned in the following text (if no diseases are mentioned, output {{"None"}}):
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 3]
```diff
@@ -106,13 +153,106 @@ templates:
...
+      {% endif %}
+
+      {% endfor %}
+
+      {{diseases.list | join | trim(",") if (diseases.list | length) > 0 else "None"}}'
+    metadata: !TemplateMetadata
+      choices_in_prompt: false
+      metrics:
+      - Accuracy
+      - Other
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:169]:
```suggestion
      - Other
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 4]
```diff
@@ -106,13 +153,106 @@ templates:
...
+      original_task: true
+    name: list_diseases_mentioned_after_text
+    reference: Ask the model to list diseases present in the text
+  ecd88889-84fa-4e28-ac0f-3bc1564e838b: !Template
+    answer_choices: null
+    id: ecd88889-84fa-4e28-ac0f-3bc1564e838b
+    jinja: 'List the diseases mentioned in the following text:
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:176]:
```suggestion
    jinja: 'List the diseases mentioned in the following text (write {{"None"}} if no disease is mentioned):
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 5]
```diff
@@ -106,13 +153,106 @@ templates:
...
+      {% endif %}
+
+      {% endfor %}
+
+      {{diseases.list | join | trim(",") if (diseases.list | length) > 0 else "None"}}'
+    metadata: !TemplateMetadata
+      choices_in_prompt: false
+      metrics:
+      - Accuracy
+      - Other
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:215]:
```suggestion
      - Other
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 6]
```diff
@@ -150,12 +290,13 @@ templates:
...
-      diseases found!"}}'
+      {{diseases.list | join | trim(",") if (diseases.list | length) > 0 else "None"}}'
     metadata: !TemplateMetadata
-      choices_in_prompt: null
-      metrics: []
-      original_task: false
+      choices_in_prompt: false
+      metrics:
+      - Accuracy
+      - Other
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:261]:
```suggestion
      - Other
```

File: promptsource/templates/ncbi_disease/templates.yaml [hunk 7]
```diff
@@ -37,12 +37,13 @@ templates:
...
```
Comment (VictorSanh) [inline promptsource/templates/ncbi_disease/templates.yaml:45]:
```suggestion
      - Other
```

**Review by VictorSanh — APPROVED** [review]
thanks for workingo n this @JetRunner !

I removed two templates (one was buggy and I didn't have the courage to fix it, the second was about manipulating indexes which we decided to exclude)

I am making a few changes to metrics: no accuracy (for ner the standard metrics is precidion/recall/f1), and "generate None" in the instructions

**Comment by JetRunner** [comment]
Sounds good!

**Pull request merged**